### ISSUE TITLE
Simplenews & Profile modules - error message

### ISSUE THREAD

**Issue by amilenkov** [backdrop-contrib/simplenews#6]
I posted this first at Backdropcms Forum
https://forum.backdropcms.org/forum/simplenews-profile-modules-error-message
--------------------

I am developing a site of the social network type, in which those interested in a given content type publications can receive messages for new posts of that content type through the simplenews module. At that old stage everything worked ok.

Yesterday, however, I added a Profile module so that registered users can add personal information so that their client account becomes their personal page.

After that I created a new profile (actually renamed the Label to the default profile "main" to be "Author profile") so that only users with the "Author" role have such an account and add some fields to that profile for CV, personal photo, links to account's social network pages or personal website, etc.

At the moment, only I can create new users on the site.

When I register a new user with the role "Author" in the registration form, the fields of the profile "Author profile" appear as expected and the registration completes successfully.

However, when I try to make any change to the client account and do not even make any changes, but only open the account for editing and save without change, the following error occurs. It does not matter whether I edit the fields defined by the profile or the standard fields of the user module - it appears even without any edit of some field:

![simplenews-error-message](https://user-images.githubusercontent.com/14857883/136196889-c19ad546-a65d-4b67-8659-d4947ca794a8.png)

Despite what is written in the message "The changes have been saved" no changes are saved and a white screen appears with this message. Access to other content on the site continues normally, it is simply not possible to edit user accounts with this role for which there is an profile.

It is not possible to edit a user account with an active role that has an activated profile.

When changing the role of a user account with one for which there is no active profile, for example from "Author" to "Authenticated" - then editing the user account is done without problems and error messages.

It is very strange that, as expected, the fields defined in the profile disappear in the user account edit form, but continue to appear when viewing the account, even though this account already has a role that should have no profile and corresponding profile's fields. Accordingly, it is not possible to edit these fields.

I found a temporary solution - when I want to edit the fields in the profile of a given account, I temporarily deactivate the simplenews module and when I finish editing profile's fields I activate it again.

However, if this issue is not resolved, I cannot authorize users to edit their accounts / profiles themselves.

Both modules, simplenews and profiles separately work perfectly, but somehow cannot work together at lest in this website.

I would be grateful for advice on how to solve the problem or idea to replace simplenews with another similar module that allows a notification subscription when there is a new post of a certain type of content, including a subscription from anonymous visitors.

**Comment by alanmels** [comment]
Hello,

I'm afraid we don't have time to troubleshoot this for now. Are you a coder? Can you enable Devel module and find what exactly triggers the error?

**Comment by amilenkov** [comment]
Thank you very much for your responsiveness!

I'm not a coder, but I've been a Drupal user since Drupal 5 and I mainly develop web design and my own Drupal and Backdrop themes for my clients. I have been working with Backdrop CMS since 2016 and now almost all my old sites and all new ones are with Backdrop CMS. I am mainly a graphic designer, I have a very basic knowledge of PHP.

But I will install Devel and try to learn how to use it. If I come up with any useful results, I will post them here.

Thanks again for the answer, now I understand that I have to expect a solution to the issue in the future, and for now I will solve the current task (messages for new publications and personal profiles of authors on the site) through some combination of other modules and techniques - Backdrop, as previously Drupal, offer vast resources for ingenuity and many different ways to achieve the desired functionality in a website.

**Comment by docwilmot** [comment]
Try adding this to the top of the function `simplenews_subscriptions_account_form()`:
```
function simplenews_subscriptions_account_form(&$form, &$form_state, $subscriber) {
  form_load_include($form_state, 'inc', 'simplenews', 'simplenews.subscription');
```

**Comment by alanmels** [comment]
Please also report this to Profile module as it's not clear what exactly triggers the error.

**Comment by amilenkov** [comment]
@docwilmot 

I tried to add this code, I found the function in the simplenews module (simplenews.subscription.inc), but either I do it incorrectly or it is not enough - the error remains.

@alanmels 

I installed Devel but it's too complicated for me, I can't handle it.

I worked for a few hours and may me all day trying to solve the problem, here's what I learned further: I uninstalled and reinstalled the Profile module. I created a new Author profile and linked it to the Author user role. I added a field for long text - the profile is edited and saved without error. However, in order for the text to be visible, this must be set in the corresponding Profile Display mode.

Only with this field - for long text, there is no error message.

However, when I add an image field, the same error reappears, despite any changes to the field settings I tried. (The Enable user pictures field in the default account settings is turned off).

After all these attempts, I give up and will create profiles of my authors not with the Profile module, but with a separate type of content with the necessary fields + some combination of taxonomy and views. I'm sorry, I can't do it anymore. Both Simplenews and Profile work well on their own, but in combination something happens and this error occurs. And I repeat, it may be useful only if I add an image field to the profile, with a plain text field no error occurs.

In order not to duplicate content, I will make a report this to Profile module with a link to this correspondence.

**Comment by docwilmot** [comment]
Sorry, typo, it should be `  form_load_include($form_state, 'inc', 'simplenews', 'includes/simplenews.subscription');`

**Comment by amilenkov** [comment]
@docwilmot

Thank you very much, the new code works great this time and the error message is gone. I worked intensively for several hours with the Profile module, added and configured various fields, tested in real work in many ways - no error messages and everything works fine.

I attach a copy of the edited file, the new code is added in line 53.

[simplenews.subscription.inc.zip](https://github.com/backdrop-contrib/simplenews/files/7317375/simplenews.subscription.inc.zip)

**Comment by alanmels** [comment]
I would really appreciate if someone could verify the solution and create a PR, as I am lacking time even to do so myself.

**Comment by amilenkov** [comment]
Hello,

As I wrote, I'm not a coder. I don't know if there are any special requirements, special order and procedures to do this - "if someone could verify the solution" as far as I tested it on my site, with this edition of simplenews.subscription.inc the problem is removed – (no error messages, both Simplenews and Profile modules work normally simultaneously and everything works OK for me at this website). 

I would try to do PR too, but I don't know how right it is since the idea and code was actually offered by @docwilmot.

I have little experience in this, last year I did a successful PR to improve the transliteration of URLs and file names in Bulgarian and all checks were completed successfully. I may be formulating the task incorrectly, but if possible I would try to help in testing and doing PR according to my modest knowledge and skills.
